core/src/main/java/org/polypheny/db/sql/ddl/altermaterializedview/SqlAlterMaterializedAddIndex.java

```diff
@@ -0,0 +1,165 @@
+import org.polypheny.db.ddl.exception.IndexExistsException;
+import org.polypheny.db.ddl.exception.MissingColumnPlacementException;
+import org.polypheny.db.ddl.exception.UnknownIndexMethodException;
+import org.polypheny.db.jdbc.Context;
+import org.polypheny.db.sql.SqlIdentifier;
+import org.polypheny.db.sql.SqlNode;
+import org.polypheny.db.sql.SqlNodeList;
+import org.polypheny.db.sql.SqlUtil;
+import org.polypheny.db.sql.SqlWriter;
+import org.polypheny.db.sql.ddl.SqlAlterMaterializedView;
+import org.polypheny.db.sql.parser.SqlParserPos;
+import org.polypheny.db.transaction.Statement;
+import org.polypheny.db.transaction.TransactionException;
+import org.polypheny.db.util.ImmutableNullableList;
+
+public class SqlAlterMaterializedAddIndex extends SqlAlterMaterializedView {
+
+
+    private final SqlIdentifier table;
+    private final SqlIdentifier indexName;
+    private final SqlIdentifier indexMethod;
+    private final SqlNodeList columnList;
+    private final boolean unique;
+    private final SqlIdentifier storeName;
+
+
+    public SqlAlterMaterializedAddIndex(
+            SqlParserPos pos,
+            SqlIdentifier table,
+            SqlNodeList columnList,
+            boolean unique,
+            SqlIdentifier indexMethod,
+            SqlIdentifier indexName,
+            SqlIdentifier storeName ) {
+        super( pos );
+        this.table = Objects.requireNonNull( table );
+        this.columnList = Objects.requireNonNull( columnList );
+        this.unique = unique;
+        this.indexName = indexName;
+        this.indexMethod = indexMethod;
+        this.storeName = storeName;
+    }
+
+
+    @Override
+    public List<SqlNode> getOperandList() {
+        return ImmutableNullableList.of( table, columnList, storeName, indexMethod, indexName );
+    }
+
+
+    @Override
+    public void unparse( SqlWriter writer, int leftPrec, int rightPrec ) {
+        writer.keyword( "ALTER" );
+        writer.keyword( "TABLE" );
+        table.unparse( writer, leftPrec, rightPrec );
+        writer.keyword( "ADD" );
+        if ( unique ) {
+            writer.keyword( "UNIQUE" );
+        }
+        writer.keyword( "INDEX" );
+        indexName.unparse( writer, leftPrec, rightPrec );
+        writer.keyword( "ON" );
+        columnList.unparse( writer, leftPrec, rightPrec );
+        if ( indexMethod != null ) {
+            writer.keyword( "USING" );
+            indexMethod.unparse( writer, leftPrec, rightPrec );
+        }
+        if ( storeName != null ) {
+            writer.keyword( "ON" );
+            writer.keyword( "STORE" );
+            storeName.unparse( writer, leftPrec, rightPrec );
+        }
+    }
+
+
+    @Override
+    public void execute( Context context, Statement statement ) {
+
+        CatalogTable catalogTable = getCatalogTable( context, table );
+
+        if ( catalogTable.isView() ) {
+            throw new RuntimeException( "Not possible to use ALTER TABLE with Views." );
```

Comment:
Adjust error message, we are not in ALTER TABLE

Reply:
Adjusted all error message for ALTER MATERIALIZED VIEWs.